src/components/MultiHopTrade/components/TradeConfirm/TradeConfirm.tsx

```diff
@@ -89,15 +90,32 @@ export const TradeConfirm = () => {
     dispatch: walletDispatch,
   } = useWallet()
 
+  const activeStepOrDefault = useAppSelector(selectActiveStepOrDefault)
+
   useEffect(() => {
     // WARNING: do not remove.
     // clear the confirmed quote on dismount to prevent stale data affecting the selectors
     return () => {
+      // TODO(gomes): This is wrong. This isn't only working on unmount as the intent suggests.
+      // This is due to the way those react-router routes work. This is actually fired on mount, voiding the stated guarantees.
+      // We will most likely want to move all these "WARNING: do not remove" effects up to the router-level, so they are *actually* fired on trade routes unmount
       dispatch(tradeQuoteSlice.actions.resetConfirmedQuote())
+
+      if (activeStepOrDefault > 0) dispatch(tradeQuoteSlice.actions.resetActiveStep())
```

Comment:
Similar rationale with this guy - we shouldn't have to check for the active step we're currently in to reset it, we should just be able to proper reset it on unmount, however, the mount/unmount with react-router route components isn't the usual one, and this doesn't run only on unmounts